views/quiz.py

```diff
@@ -0,0 +1,307 @@
+        "responses": [
+            "Time's up, sweetie! You didn't answer within the given time. But it's okay, let's try the next one!",
+            "It seems like you ran out of time to answer this one, darling. Don't worry, there's always next time!",
+            "Unfortunately, time has run out and I didn't receive an answer from you, honey.",
+            "Looks like you didn't manage to answer this one within the time limit, sweetheart.",
+            "Time has run out for this question and I didn't get your response, darling.",
+            "Unfortunately, you didn't manage to respond within the given time frame for this one, sweetie.",
+            "Looks like time ran out before you could respond to this one, honey. But don't worry, there's always a next time!"
+        ]
+    },
+    "next_question": [
+        'The next question will be {next}, sweetie! Are you ready?',
+        'Get ready for the next question {next}, darling! Let\'s do this together!',
+        'The next question is coming up {next}, honey! I\'m excited to see your answer!',
+        'Stay tuned for the next question {next}, sweetheart! I believe in you!'
+    ]
+}
+
+class AnswerModal(discord.ui.Modal):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(title="Enter your answer", *args, **kwargs)
+        self.answer: str = ""
+
+        self.add_item(
+            discord.ui.TextInput(
+                label="Answer",
+                style=discord.TextStyle.long,
+            )
+        )
+
+    async def on_submit(self, interaction: discord.Interaction) -> None:
+        await interaction.response.defer()
+        self.answer = self.children[0].value
+        self.stop()
+
+class ResetAttemptView(discord.ui.View):
+    def __init__(self, ctx: commands.Context, user_data: dict[str, Any], timeout: float = 30):
+        super().__init__(timeout=timeout)
+
+        self.ctx: commands.Context = ctx
+        self.data: dict[str, Any] = user_data
+        self.response: discord.Message = None
+
+    async def on_timeout(self) -> None:
+        for child in self.children:
+            child.TextStyle = discord.ButtonStyle.grey
+            child.disabled = True
+        
+        if self.response:
+            await self.response.edit(view=self)
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        return interaction.user == self.ctx.author
+
+    @discord.ui.button(label="Buy", emoji="🛍️", style=discord.ButtonStyle.green)
+    async def buy(self, interaction: discord.Interaction, button: discord.ui.Button) -> None:
+        if self.data.get("candies") < 50:
+            await interaction.response.send_message("You do not have enough candies to initiate the reset.!", ephemeral=True)
+            return await self.on_timeout()
+        
+        await func.update_user(self.ctx.author.id, {
+            "$set": {"game_state.quiz_game.attempted.times": 0},
+            "$inc": {"candies": -50}
+        })
```

Comment:
Could keep the 50 in a variable in game config or somewhere instead of hardcoding it here.